app/services/foreman/renderer/scope/macros/snippet_rendering.rb

```diff
@@ -22,7 +38,16 @@ module SnippetRendering
             see 'snippet', description: 'Snippet#snippet', scope: Foreman::Renderer::Scope::Macros::SnippetRendering
           end
           def snippet_if_exists(name, options = {})
-            snippet(name, { silent: true }, variables: options[:variables] || {})
+            tpl_name = ''
+            if name.is_a?(Array)
+              name.each do |tpl|
+                tpl_name = tpl
+                break if snippet_exists(tpl_name)
+              end
```

Comment:
This is probably better, but there's a difference. In your case even if the snipped didn't exist, it would still be set to the last entry of `name`. In this case it's set to `nil`. I don't know if `snippet` can deal with that.
```suggestion
              tpl_name = name.find { |tpl| snippet_exists(tpl_name) }
```

Alternative is:
```suggestion
              tpl_name = name.find { |tpl| snippet_exists(tpl_name) } || ''
```

Then you can drop line 41.